config/server.go

```diff
@@ -68,9 +72,29 @@ func LoadServerFromEnv() *Server {
 	var server Server
 	LoadEnvConfig(&server)
 	if server.HTTPPort != 0 || server.RPCPort != 0 ||
-		server.HTTPAccessLog != "" || server.RPCAccessLog != "" ||
+		server.HTTPAccessLog != nil || server.RPCAccessLog != nil ||
 		server.HealthCheckType != "" || server.HealthCheckPath != "" {
 		return &server
 	}
 	return nil
 }
+
+// NewAccessLogMiddleware will wrap a logrotate-aware Apache-style access log handler
+// around the given http.Handler if an access log location is provided by the config,
+// or optionally send access logs to stdout.
+func NewAccessLogMiddleware(logLocation *string, handler http.Handler) (http.Handler, error) {
+	if logLocation == nil {
+		return handler, nil
+	}
+	var lw io.Writer
+	var err error
+	if *logLocation == "stdout" {
+		lw = os.Stdout
+	} else {
+		lw, err = logrotate.NewFile(*logLocation)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return handlers.CombinedLoggingHandler(lw, handler), nil
```

Comment:
Mind tossing this in a `switch *logLocation {...` block?  It'd clean it up a tad and make it real easy and clear to add in "stderr" as well.